<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>600</height>
   </rect>
  </property>
  <property name="toolTipDuration">
   <number>5</number>
  </property>
  <widget class="QWidget" name="central_widget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>1001</width>
      <height>601</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <layout class="QVBoxLayout" name="container">
     <property name="leftMargin">
      <number>7</number>
     </property>
     <property name="topMargin">
      <number>7</number>
     </property>
     <property name="rightMargin">
      <number>7</number>
     </property>
     <property name="bottomMargin">
      <number>7</number>
     </property>
     <item>
      <widget class="QLabel" name="name">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="font">
        <font>
         <family>Consolas</family>
         <pointsize>24</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Treaps</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="coordinator">
       <item>
        <widget class="QPushButton" name="about_treap_button">
         <property name="text">
          <string>What is the &quot;Treap&quot;?</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="about_app_button">
         <property name="text">
          <string>How does the app work?</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="log_button">
         <property name="text">
          <string>Log</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="first_spacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="save_button">
         <property name="text">
          <string>Save</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="load_button">
         <property name="text">
          <string>Load</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="reset_button">
         <property name="text">
          <string>Clear History</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="second_spacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="back_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Back</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="step_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Forward</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="input_container">
       <item>
        <widget class="QPushButton" name="generator_button">
         <property name="text">
          <string>Generate input</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="line_input">
         <property name="font">
          <font>
           <family>Consolas</family>
          </font>
         </property>
         <property name="placeholderText">
          <string>tree members or filename</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="line_button">
         <property name="font">
          <font>
           <family>Consolas</family>
          </font>
         </property>
         <property name="text">
          <string>Build from line</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="file_button">
         <property name="font">
          <font>
           <family>Consolas</family>
          </font>
         </property>
         <property name="text">
          <string>Build from file</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="step_box">
         <property name="text">
          <string>Step by step</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="type_selector"/>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="font">
        <font>
         <family>Consolas</family>
         <pointsize>7</pointsize>
        </font>
       </property>
       <property name="text">
        <string>NB! Input must be formatted. Trees must be divided with &quot; = &quot;, leaves with  &quot;, &quot;. In multiple mode ranges are also available with delimiter &quot; .. &quot;. See more in &quot;How does this app work?&quot;.</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="canvases">
       <item>
        <widget class="MyGLWidget" name="canvas1">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="MyGLWidget" name="canvas2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="output_container">
       <item>
        <widget class="QLabel" name="answer">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>Consolas</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="merge_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="font">
          <font>
           <family>Consolas</family>
          </font>
         </property>
         <property name="text">
          <string>Merge</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="split_pos_box">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="split_button">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Split</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="multiple_button">
         <property name="font">
          <font>
           <family>Consolas</family>
          </font>
         </property>
         <property name="text">
          <string>Multiple times</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="subscription">
       <property name="font">
        <font>
         <family>Consolas</family>
         <pointsize>7</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Developed by group 8381 student A. Sergeev.</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">myglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
